zrml/court/src/types.rs

```diff
@@ -0,0 +1,375 @@
+// Copyright 2021-2022 Zeitgeist PM LLC.
+//
+// This file is part of Zeitgeist.
+//
+// Zeitgeist is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at
+// your option) any later version.
+//
+// Zeitgeist is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
+extern crate alloc;
+use alloc::{vec, vec::Vec};
+use zeitgeist_primitives::types::OutcomeReport;
+
+/// The type of the court identifier.
+pub type CourtId = u128;
+
+/// The different court vote types. This can be extended to allow different decision making options.
+#[derive(
+    parity_scale_codec::Decode,
+    parity_scale_codec::Encode,
+    parity_scale_codec::MaxEncodedLen,
+    scale_info::TypeInfo,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+)]
+pub enum VoteItemType {
+    Outcome,
+    Binary,
+}
+
+/// The different court vote types with their raw values.
+/// This can be extended to allow different decision making options.
+#[derive(
+    parity_scale_codec::Decode,
+    parity_scale_codec::Encode,
+    parity_scale_codec::MaxEncodedLen,
+    scale_info::TypeInfo,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    Ord,
+    PartialOrd,
+)]
+pub enum VoteItem {
```

Comment:
For now we use court to resolve on disputes. This was the requirement. I think it is a plus that we have foreseen the future use of court to evolve into a general stake-weighted decision machine. It's easier now to extend other decisions. I think it is overkill to properly implement the general decision machine in the first iteration.

We should keep in mind that we also have to sacrifice. The code is never perfect. In reality there is always room for improvement. For now we should focus on getting things done I guess. The proper implementation of general decisions (like full binary decision with integration) is not part of the current requirements. It's a nice to have if I already have a general way `VoteItem` to make a later integration easier. 

Let us create an issue for the extension of court for `delay market` and other purposes.

Reply:
In addition to that, another API is required, if you were to allow the usage to other users of the ecosystem.

I will try to implement this on another branch, since I don't know yet how much I need to change for this.

But first, let me work on fixing the documentation PR on the other repository, since this is more urgent atm. 

I also think the randomness pallet is more urgent too, before I get to making the pallet generic.